### ISSUE TITLE
Instructor: Feedback Session: Mark mandatory fields for creating and editing session

### ISSUE THREAD

**Issue by mg14777** [TEAMMATES/teammates#8931]
**Description of feature/enhancement**
Mark mandatory fields using `*` or some other mechanism when user creates a new feedback session or edits existing feedback session.

Currently, user is only notified of mandatory fields on form submission

![image](https://user-images.githubusercontent.com/11634699/42173382-bc790c44-7e16-11e8-99b4-86db3aefccce.png)

It will be even better if these errors are attached to the fields rather being listed at one place.
Example:
![image](https://user-images.githubusercontent.com/11634699/42173513-2ed03754-7e17-11e8-8236-6a6cba48f14c.png)

**Comment by sukanta-27** [comment]
We can also use HTML 5 form validation here. Making the mandatory fields as `required`?

**Comment by mg14777** [comment]
> We can also use HTML 5 form validation here

One slight issue we need to keep in mind is that different browsers tend to have different behaviour when displaying the error. We might need some additional CSS to ensure consistency.

**Comment by sukanta-27** [comment]
>We might need some additional CSS to ensure consistency.

Yes, we can do that, but as this is a browser specific behaviour, do we need to deal with this? I mean, if our main concern is that, the user doesn't leave a important field blank, then isn't adding additional CSS an overkill? But yes, if we do go this way, there are many places where we can use the CSS for better UX.

**Comment by mg14777** [comment]
@sukanta-27 Yeah I mainly meant from a UX perspective. It's not core to the issue at hand.

**Comment by damithc** [comment]
We try to avoid/minimize front-end validation (in spite of the UX cost) because we still have to do back-end validation (reason: front-end validation can be bypassed) and then we end up duplicating the validation logic, once in JS and once in Java.

**Comment by ah91086** [comment]
Given what damithc said, would the best course of action be to simply edit the corresponding .tag file to mark the mandatory fields with a star since front-end validation isn't an option?

**Comment by LiHaoTan** [comment]
Minimal frontend validation can still be used with HTML form validation but more important validation checks should only likely exist in the backend.

There are certainly UX issues like what @mg14777 mentioned, including accessibility. However, these are things that we do not have to resolve on our own, for example, Bootstrap 4 supports HTML form validation.

**Comment by gildas15** [comment]
PLease can you assign this issue to me. I want to work on it

**Comment by Ting-TT** [comment]
Hii, just wondering if this issue is still open to fix?

It behaves like this at the moment:

<img width="941" alt="Screenshot 2023-10-20 at 6 47 50 pm" src="https://github.com/TEAMMATES/teammates/assets/111097257/3c7a45a8-9b48-4918-9435-4a0c20d2b3c7"> <br>
  
Maybe I can try to modify it to make it behave similar like this: 


<img width="941" alt="Screenshot 2023-10-20 at 6 55 39 pm" src="https://github.com/TEAMMATES/teammates/assets/111097257/0a297a9b-e343-4889-809e-69208419ed72">
